src/Microsoft.TemplateEngine.Core/Expressions/Cpp2/Cpp2StyleEvaluatorDefinition.cs

```diff
@@ -0,0 +1,349 @@
+
+            if (isFaulted)
+            {
+                faulted = true;
+                return false;
+            }
+
+            try
+            {
+                object evalResult = result.Evaluate();
+                bool r = (bool)Convert.ChangeType(evalResult, typeof(bool));
+                faulted = false;
+                return r;
+            }
+            catch
+            {
+                faulted = true;
+                return false;
+            }
+        }
+
+        public static bool EvaluateFromString(string text, IVariableCollection variables)
+        {
+            using (MemoryStream ms = new MemoryStream(Encoding.UTF8.GetBytes(text)))
+            using (MemoryStream res = new MemoryStream())
+            {
+                EngineConfig cfg = new EngineConfig(variables);
+                IProcessorState state = new ProcessorState(ms, res, (int)ms.Length, (int)ms.Length, cfg, NoOperationProviders);
+                int len = (int)ms.Length;
+                int pos = 0;
+                bool faulted;
+                return Cpp2StyleEvaluator(state, ref len, ref pos, out faulted);
+            }
+        }
+
+        private static int? AttemptComparableComparison(object left, object right)
+        {
+            IComparable ls = left as IComparable;
+            IComparable rs = right as IComparable;
+
+            if (ls == null || rs == null)
+            {
+                return null;
+            }
+
+            return ls.CompareTo(rs);
+        }
+
+        private static int? AttemptLexographicComparison(object left, object right)
+        {
+            string ls = left as string;
+            string rs = right as string;
+
+            if (ls == null || rs == null)
+            {
+                return null;
+            }
+
+            return string.Compare(ls, rs, StringComparison.OrdinalIgnoreCase);
+        }
+
+        private static int? AttemptNumericComparison(object left, object right)
+        {
+            bool leftIsDouble = left is double;
+            bool rightIsDouble = right is double;
+            double ld = leftIsDouble ? (double)left : 0;
+            double rd = rightIsDouble ? (double)right : 0;
+
+            if (!leftIsDouble)
+            {
+                string ls = left as string;
+
+                if (ls != null)
+                {
+                    int lh;
+                    if (double.TryParse(ls, out ld))
+                    {
+                    }
+                    else if (ls.StartsWith("0x", StringComparison.OrdinalIgnoreCase) && int.TryParse(ls.Substring(2), NumberStyles.HexNumber, CultureInfo.InvariantCulture, out lh))
+                    {
+                        ld = lh;
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                }
+            }
+
```

Comment:
This method is difficult to follow due to the short variable names. 
Also, the code within the "if (!leftIsDouble)" and within (!rightIsDouble) is duplicated.